sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -77,9 +77,12 @@ object AggUtils {
       child: SparkPlan): SparkPlan = {
     val useHash = Aggregate.supportsHashAggregate(
       aggregateExpressions.flatMap(_.aggregateFunction.aggBufferAttributes))
+
+    val forceHashAggregate = forceApplyHashAggregate(child.conf)
```

Comment:
I don't think we need this config. HashAggregateExec is the first choice. We will always pick it if the query is applicable.

Reply:
I did it because I like that this way it is immediately obvious in the test case what the config is saying instead of relying on a default which could technically change at any time.

However if you disagree I'm not against removing it